hax/hax/util.py

```diff
@@ -207,6 +207,28 @@ def to_payload(v: TxPutKV) -> Dict[str, Any]:
         except (ConsulException, HTTPError, RequestException) as e:
             raise HAConsistencyException('Failed to put value to KV') from e
 
+    @repeat_if_fails(max_retries=5)
+    def kv_delete_in_transaction(self, tx_payload: List[KeyDelete]) -> bool:
+        def to_payload(v: KeyDelete) -> Dict[str, Any]:
+            if v.recurse is True:
```

Comment:
`if v.recurse:` is sufficient.

`is` operator compares the identity (memory address) of two objects. Perhaps `True` and `False` objects are kind of singletons within Python interpreter which makes your code working, but in general case this habit is a bit dangerous.

And I guess these four lines can be replaced by ternary operator like this (please double check, I didn't try it in REPL):
```suggestion
            return {'KV': {'Key': v.name, 'Verb': 'delete-tree' if v.recurse else 'delete'}}
```

Reply:
DOne